include/nbl/builtin/hlsl/shapes/spherical_triangle.hlsl

```diff
@@ -0,0 +1,132 @@
+// Copyright (C) 2018-2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_SHAPES_SPHERICAL_TRIANGLE_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SHAPES_SPHERICAL_TRIANGLE_INCLUDED_
+
+#include <nbl/builtin/hlsl/tgmath.hlsl>
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/limits.hlsl>
+#include <nbl/builtin/hlsl/math/functions.hlsl>
+#include <nbl/builtin/hlsl/math/angle_adding.hlsl>
+#include <nbl/builtin/hlsl/numbers.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace shapes
+{
+
+template<typename T>
+struct SphericalTriangle
+{
+    using scalar_type = T;
+    using vector3_type = vector<T, 3>;
+
+    static SphericalTriangle<T> create(NBL_CONST_REF_ARG(vector3_type) vertex0, NBL_CONST_REF_ARG(vector3_type) vertex1, NBL_CONST_REF_ARG(vector3_type) vertex2, NBL_CONST_REF_ARG(vector3_type) origin)
+    {
+        SphericalTriangle<T> retval;
+        retval.vertex0 = nbl::hlsl::normalize(vertex0 - origin);
+        retval.vertex1 = nbl::hlsl::normalize(vertex1 - origin);
+        retval.vertex2 = nbl::hlsl::normalize(vertex2 - origin);
+        retval.cos_sides = vector3_type(hlsl::dot(retval.vertex1, retval.vertex2), hlsl::dot(retval.vertex2, retval.vertex0), hlsl::dot(retval.vertex0, retval.vertex1));
+        const vector3_type csc_sides2 = hlsl::promote<vector3_type>(1.0) - retval.cos_sides * retval.cos_sides;
```

Comment:
nitpick, this should be called `sin_sides2` https://github.com/Devsh-Graphics-Programming/Nabla/pull/959#discussion_r2588664552